### PR TITLE
Add new analytics events, introduce /vote endpoint

### PR DESCRIPTION
**Pull request by calyptobai**

### CODE REVIEW

**Comment by anastasiya1155** [comment]
When is the identify call made? Preferably should be done after github login or on launch

**Comment by anastasiya1155** [comment]
For upvote route we need query_id, but FE doesn't know it. It is not present in answer stream

File: server/bleep/src/webserver/answer.rs [hunk 1]
```diff
@@ -44,6 +44,30 @@ use response::{Exchange, SearchResult, SearchStep, Update};
...
+#[derive(Clone, Debug, serde::Deserialize)]
+pub struct Vote {
+    pub positive: bool,
+    pub thread_id: uuid::Uuid,
+    pub query_id: uuid::Uuid,
+    pub repo_ref: Option<RepoRef>,
+}
+
+pub(super) async fn vote(
+    Query(params): Query<Vote>,
```
Comment (anastasiya1155) [inline server/bleep/src/webserver/answer.rs:56]: we shouldn't use query params for POST requests, we should parse request body instead

File: server/bleep/src/webserver/answer.rs [hunk 2]
```diff
@@ -44,6 +44,30 @@ use response::{Exchange, SearchResult, SearchStep, Update};
 
 const TIMEOUT_SECS: u64 = 60;
 
+#[derive(Clone, Debug, serde::Deserialize)]
+pub struct Vote {
+    pub positive: bool,
+    pub thread_id: uuid::Uuid,
+    pub query_id: uuid::Uuid,
+    pub repo_ref: Option<RepoRef>,
+}
```
Comment (anastasiya1155) [inline server/bleep/src/webserver/answer.rs:53]: also, should accept and track in analytics comment (for downvotes)

**Review by anastasiya1155 — CHANGES_REQUESTED** [review]

**Comment by calyptobai** [comment]
@anastasiya1155 The identify call is now made on first login. Do we want it to also occur on a specific route, for users which are already logged in? It should now include the parameters you noted, though `user_id` *is* `user_name`.

Additionally:

> For upvote route we need query_id, but FE doesn't know it. It is not present in answer stream

The first message sent back from `/answer` has been changed from a thread ID string to an initialization object of the form:

```json
{
    "thread_id": "...",
    "query_id": "...",
}
```

> also, should accept (and track in analytics) the user comment left when downvoting the response

Now, `.../vote` replaces `positive: boolean` with a `feedback` parameter in the forms:

```json
{"type":"positive"}
```
```json
{"type":"negative","feedback":"..."}
```

**Comment by anastasiya1155** [comment]
I still don't see the 'identify' call in the analytics panel. Are you sure it is triggered in Tauri? As far as I remember login flow on the cloud and in Tauri differs.

**Comment by anastasiya1155** [comment]
I don't see any events for repo indexed/removed in the analytics console. fyi the FE uses `delete /repos/indexed` and `get /repos/sync` endpoints for managing repos

**Comment by calyptobai** [comment]
> I still don't see the 'identify' call in the analytics panel. Are you sure it is triggered in Tauri? As far as I remember login flow on the cloud and in Tauri differs.

We should now track successful logins with `/remotes/github/login`.

> I don't see any events for repo indexed/removed in the analytics console. fyi the FE uses `delete /repos/indexed` and `get /repos/sync` endpoints for managing repos

Deletions are now tracked via `DELETE /repos/indexed`. Addition is tracked as part of `PUT /repos/indexed`.

**Comment by calyptobai** [comment]
> We need to do some tweaks to identify call:
> 
> * device_id shouldn't have a trailing newline (previously was trimmed on the FE)
> * user_id should be the same as tracking_id from /config endpoint previously used by FE
> * github_username should be in traits

> I don't see events for repo adding/removing in the analytics console

These should now be addressed.

**Review by anastasiya1155 — APPROVED** [review]

**Pull request merged**